1_fetch/src/get_sb_data.R

```diff
@@ -206,18 +72,95 @@ download_children <-function(sites, sb_table_reduced, dldir, workdir, outdir, ou
   
   itemfails <- as.data.frame(itemfails)
   
-  filepath1 <- file.path(outdir, out_file_name)
+  filepath1 <- paste0(outdir, "/", out_file_name, item, ".csv")
   write_csv(data_at_sites, filepath1)
-  filepath2 <- file.path(outdir, paste0("FAILS_", out_file_name))
+  filepath2 <- paste0(outdir, "/", out_file_name, item, "_FAILS.csv")
   write_csv(itemfails, filepath2)
   return_df = c(filepath1, filepath2)
   return(return_df)
 }
 
-calc_avg_dams <- function(sb_data) {
-  sb_data <- read_csv(sb_data, show_col_types = FALSE) %>%
-    suppressMessages()
-  dams <- sb_data %>%
+get_sb_data_log_g2 <- function(sb_var_ids, file_out) {
+  
+  sb_log <- tibble(sb_id = character(), 
+                   last_update = character())
+  for (i in 1:nrow(sb_var_ids)) {
+    sb_id <- as.character(sb_var_ids[i,1])
+    last_update <- item_get_fields(sb_id, "provenance")$lastUpdated
+    log_item <- tibble(sb_id, last_update)
+    sb_log <- bind_rows(sb_log, log_item)
+  }
+  
+  write_csv(sb_log, file_out)
+  return(file_out)
+  
+}
+
+prep_feature_vars <- function(sb_var_data, sites) {
```

Comment:
Can you comment what columns are expected to exist in both of these variables? That will help us when we want to apply this to ungaged reaches

Reply:
This is a good idea - going through this exercise made me realize that the function requires these variables from the Gages2.1 dataset, which will not be available everywhere: npdes, fwwd, strg, devl, cndp. What are we going to do about that, just drop these from our features list?

Reply:
I'm keeping those in for modeling for now, but if they're not available everywhere (ungaged locations) we would need to drop them. I think there are some overlaps with the NHD attributes, but they are not exactly the same.

Reply:
Added an argument to list out additional variables (columns) to keep from the sites argument (in this case, Gages2.1).